Doc/library/fcntl.rst

```diff
@@ -14,8 +14,8 @@
 ----------------
 
 This module performs file control and I/O control on file descriptors. It is an
-interface to the :c:func:`fcntl` and :c:func:`ioctl` Unix routines.  For a
-complete description of these calls, see :manpage:`fcntl(2)` and
+interface to the :c:func:`!fcntl` and :c:func:`!ioctl` Unix system calls.  For a
```

Comment:
"system calls" is not accurate. These are POSIX C library `libc` functions. Routines was an appropriately ambiguous word that is basically shorthand for library calls. _(that they are system calls behind the scenes is an OS implementation detail)_

When updating text like this I tend to replace the Unix term with POSIX as that is actually relevant, it not being the '90s and all.

---

Why not shorten the text entirely and just make these be the :manpage: links directly instead of the last sentence in the paragraph existing at all?

Reply:
Hmmm... The manpage links are to section 2, which in my old Unix lizard brain reads as "system calls." Historically, I don't think that was a behind-the-scenes thing. There are differences between system calls and library routines (performance, task switch opportunity, etc). If that's changed over time for some Unix-y operating systems, it probably hasn't changed for all of them. (Are they still actual system calls on NetBSD, for example?)

I going to speak a quiet part loud which has been bugging me as I worked through other doc files. gh-101100 is (I think) meant to clean up broken references. The first few changes I made, I restricted my attention to that task. Then came a bunch of bike shedding about content. This time I decided I would fix some wording I thought was incorrect. Now I'm supposed to change this back. Can't win for losing.